test/type-error.js

```diff
@@ -0,0 +1,105 @@
+import test from 'ava';
+import avaRuleTester from 'eslint-ava-rule-tester';
+import rule from '../rules/type-error';
+
+const ruleTester = avaRuleTester(test, {
+	env: {
+		es6: true
+	}
+});
+
+const errors = [{
+	ruleId: 'type-error',
+	message: '`new Error()` is too unspecific for a typecheck, use `new TypeError()` instead.'
+}];
+
+ruleTester.run('type-error', rule, {
+	valid: [
+		`if (Array.isArray(foo) || ArrayBuffer.isView(foo)) {
```

Comment:
Testing-wise, these valid tests are not very interesting because you don't test when `new Error()` is allowed, which is what we want to know. "Does this case forbid new Error() when I do X?"
Keep the actual tests, but I'd like to see new Error() thrown in more diverse conditions. Inside a condition, outside one, inside a loop, at the top-level, thrown with a custom error or, with non-type checking conditions, ...